### PR TITLE
#17554-Fix-for-MySQL-username-with-proper-back-quotes

### PR DESCRIPTION
**Pull request by amargulaganji**
Fix for MySQL username isn't with proper back quotes when dragging/copying to SQL editor #17554

[#17554](https://github.com/dbeaver/dbeaver/issues/17554)

17554-Test

### CODE REVIEW

**Comment by LonwoLonwo** [comment]
Hello @amargulaganji 
Thanks for the contribution.
Yes, your fix helps with user's names.
But `quoteIdentifier` is a base method. What if `str` will be a table name? Like `R@se`? Or column name? Or catalog name? With symbol @ inside.

**Comment by amargulaganji** [comment]
Hi @LonwoLonwo,
Thank you for your observation.
As you noticed, changes to `quoteIdentifier` method will affect the table and column names.
So removed changes from `quoteIdentifier`. Made changes that will make difference to the user's name only.

Test Result:
Test-user-table-column-name

**Comment by LonwoLonwo** [comment]
That's better.

2022-11-17 14_04_37-DBeaver 22 2 5 - _localhost 3_ Script-22

But now the user's list looks something like this. Strange that I can't see it on your video.

**Comment by amargulaganji** [comment]
@LonwoLonwo Made changes that will not affect the user's list.

Test Result :
Test-username-ui

**Comment by LonwoLonwo** [comment]
Oh, no :( You can not fix the local problem on the general level

**Review by ShadelessFox — CHANGES_REQUESTED** [review]
We don't want to handle such a weird case on the top level.

Instead, just implement the `DBPQualifiedObject` interface for the `MySQLUser` class.

**Comment by amargulaganji** [comment]
Thank you @ShadelessFox  for your suggestion. It helped me to fix the issue locally.

Please review my changes.

Test Result:
17554-Test-Result

File: plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLUser.java [hunk 1]
```diff
@@ -309,4 +311,9 @@ public DBSObject refreshObject(@NotNull DBRProgressMonitor monitor) throws DBExc
         grants = null;
         return this;
     }
+
+    @Override
+    public String getFullyQualifiedName(DBPEvaluationContext context) {
+        return "`" + userName + "`@`" + host + "`";
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.mysql/src/org/jkiss/dbeaver/ext/mysql/model/MySQLUser.java]: The `userName` might contain backquotes, please use `DBUtils#getQuotedIdentifier` with `quoteAlways` set to `true`

**Review by ShadelessFox — CHANGES_REQUESTED** [review]

**Review by ShadelessFox — APPROVED** [review]

**Review by E1izabeth — APPROVED** [review]

**Comment by amargulaganji** [comment]
Hi @LonwoLonwo , Could you please review and merge?

**Comment by E1izabeth** [comment]
@amargulaganji your PR is waiting for testing.

**Comment by Matvey16** [comment]
Verified

**Pull request merged**